src/panels/lovelace/cards/hui-picture-elements-card.ts

```diff
@@ -0,0 +1,103 @@
+import { html, LitElement } from "@polymer/lit-element";
+
+import createHuiElement from "../common/create-hui-element.js";
+
+import { LovelaceCard, LovelaceConfig } from "../types";
+import { HomeAssistant } from "../../../types.js";
+import { ElementConfig } from "../elements/types.js";
+import { TemplateResult } from "lit-html";
+
+interface Config extends LovelaceConfig {
+  title?: string;
+  image: string;
+  elements: ElementConfig[];
+}
+
+class HuiPictureElementsCard extends LitElement implements LovelaceCard {
+  protected config?: Config;
+  private _hass?: HomeAssistant;
+
+  static get properties() {
+    return {
+      config: {},
+    };
+  }
+
+  set hass(hass: HomeAssistant) {
+    this._hass = hass;
+    for (const el of this.shadowRoot!.querySelectorAll("#root > *")) {
+      const element = el as LovelaceCard;
+      element.hass = this._hass;
+    }
+  }
+
+  public getCardSize(): number {
+    return 4;
+  }
+
+  public setConfig(config: Config): void {
+    if (!config) {
+      throw new Error("Invalid Configuration");
+    } else if (!config.image) {
+      throw new Error("Invalid Configuration: image required");
+    } else if (!Array.isArray(config.elements)) {
+      throw new Error("Invalid Configuration: elements required");
+    }
+
```

Comment:
We should add a check that each entry in `config.elements` has all the required props https://www.home-assistant.io/lovelace/picture-elements/#elements

Reply:
(not per type, just the base ones (type, entity, style)

Reply:
(only type and style `service-button` has no entity)

Reply:
Should that check just be part of `createHuiElement`?

Reply:
We actually already handle it. createHuiElement will render an error card if the element setConfig raises, as it should. no need to change.